### PR TITLE
#1076: nominal subtyping for ids

### LINKED ISSUE #1073: Brick headers not being generated
* A file is generated, but it's an empty list
* See: https://github.com/pixiebrix/pixiebrix-extension/actions/runs/1132779007

### LINKED ISSUE #1076: Nominal typing for ids
**Context**
- We have two different kinds of ids floating around: UUIDs (surrogate keys) and registry ids
- These are all labelled as string
- Even with type aliases, it there would be inconsistent reporting of the type in the IDE

### PR DESCRIPTION
**Pull request by twschiller**
Closes #1076 
Closes #1073

- Introduces nominal subtyping for UUID vs. RegistryIds
- Fixes brick generation: #1073 (🤦 TypeScript not blocking Object.values on Maps)
- Remove header generation code for extensionPoints, as we don't have any hard-coded extension points anymore
- Fail build if no headers are generated

### CODE REVIEW

File: src/contentScript/lifecycle.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ const WAIT_LOADED_INTERVAL_MS = 25;
 async function installScriptOnce(): Promise<void> {
   // https://stackoverflow.com/questions/9515704/insert-code-into-the-page-context-using-a-content-script/9517879#9517879
   // https://stackoverflow.com/questions/9602022/chrome-extension-retrieving-global-variable-from-webpage
-  if (!_scriptPromise) {
```
Comment (fregante) [inline src/contentScript/lifecycle.ts:47]: Why this change?

File: src/core.ts [hunk 2]
```diff
@@ -34,6 +34,20 @@ export type RenderedHTML = string;
 
 export type ActionType = string;
 
+export type UUID = string & {
+  // Nominal subtyping
+  _uuidBrand: never;
+};
```
Comment (fregante) [inline src/core.ts:40]: Too bad TypeScript doesn't support this natively yet:

- https://github.com/Microsoft/Typescript/issues/202

I think I achieved something similar with a plain string type before: 😅 

https://github.com/sindresorhus/refined-github/blob/ac8de6bd0ef0a6016732a69556e4eb9c014cfd38/source/globals.d.ts#L5

**Review by fregante — COMMENTED** [review]
👍

File: src/contentScript/lifecycle.ts (hunk 1 above)
Comment (twschiller) [reply src/contentScript/lifecycle.ts:47]: Linter was complaining about using "!" on a promise (which makes sense, as it's error-prone, since in most cases the intention would probably be using "!" on the result of the promise)

File: src/core.ts (hunk 2 above)
Comment (twschiller) [reply src/core.ts:40]: In a past life I worked on https://checkerframework.org/, which allowed you apply additional type qualifiers to types in Java. Double that TypeScript will be getting support for it anytime soon, but you could do a lot of cool things with custom types

Definitely nice having string union types, and they do have some wacky helpers e.g., for capitalizing string: https://www.typescriptlang.org/docs/handbook/utility-types.html#intrinsic-string-manipulation-types
